test/controllers/businessName/CaptureBusinessNameControllerSpec.scala

```diff
@@ -0,0 +1,227 @@
+import connectors.httpParsers.GetCustomerInfoHttpParser.GetCustomerInfoResponse
+import controllers.ControllerBaseSpec
+import org.jsoup.Jsoup
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import play.api.http.Status
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.businessName.CaptureBusinessNameView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CaptureBusinessNameControllerSpec extends ControllerBaseSpec {
+
+  def setup(result: GetCustomerInfoResponse): Any =
+    when(mockVatSubscriptionService.getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext]))
+      .thenReturn(Future.successful(result))
+
+  def target(result: GetCustomerInfoResponse = Right(customerInfoNoPending)): CaptureBusinessNameController = {
+    setup(result)
+    new CaptureBusinessNameController(mockErrorHandler, inject[CaptureBusinessNameView], mockVatSubscriptionService)
+  }
+
+  val testValidationBusinessName: String = "Current Business Name"
+  val testValidBusinessName: String      = "A Valid Business Name"
+  val testInvalidBusinessName: String    = ""
+
+  val controller = new CaptureBusinessNameController(
+    mockErrorHandler,
+    inject[CaptureBusinessNameView],
+    mockVatSubscriptionService
+  )
+
+  "Calling the show action" when {
+
+    "a user is enrolled with a valid enrolment" when {
+
+      "the user's current business name is retrieved from session" should {
+
+        lazy val result = target().show(request.withSession(validationBusinessNameKey -> testValidationBusinessName, businessNameAccessPermittedKey -> "true"))
+
+        lazy val document = Jsoup.parse(bodyOf(result))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+
+        "prepopulate the form with the user's current business name" in {
+          document.select("#business-name").attr("value") shouldBe testValidationBusinessName
+        }
+      }
+
+      "the previous form value is retrieved from session" should {
+
+        lazy val result = target().show(request.withSession(
+          validationBusinessNameKey -> testValidationBusinessName,
+          prepopulationBusinessNameKey -> testValidBusinessName,
+          businessNameAccessPermittedKey -> "true")
+        )
+        lazy val document = Jsoup.parse(bodyOf(result))
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+
+        "prepopulate the form with the previously entered form value" in {
+          document.select("#business-name").attr("value") shouldBe testValidBusinessName
+        }
+      }
+    }
```

Comment:
Could you add a test for the case where the user doesn't have validationBusinessNameKey